ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -263,4 +264,23 @@ default boolean supportsPartitionTransform() {
   default String getFileFormatPropertyKey() {
     return null;
   }
+
+  /**
+   * Check if we should use the {@link #nativeCommit(Properties, boolean)} method for committing inserts instead of
+   * using file copy in the {@link org.apache.hadoop.hive.ql.exec.MoveTask}-s.
+   * @return
+   */
+  default boolean useNativeCommit() {
+    return false;
+  }
+
+  /**
+   * Commits the inserts for the non-native tables. Used in the {@link org.apache.hadoop.hive.ql.exec.MoveTask}.
```

Comment:
It's a bit strange that we use the term native commit but only for non-native tables :) But to be honest, I couldn't find a better name yet either, because `nativeCommit` still kinda makes sense :)

Reply:
What do you think about these names?
```
  /**
   * Checks if we should keep the {@link org.apache.hadoop.hive.ql.exec.MoveTask} and use the
   * {@link #storageHandlerCommit(Properties, boolean)} method for committing inserts instead of
   * {@link org.apache.hadoop.hive.metastore.DefaultHiveMetaHook#commitInsertTable(Table, boolean)}.
   * @return Returns true if we should use the {@link #storageHandlerCommit(Properties, boolean)} method
   */
  default boolean keepMoveTask() {
    return false;
  }

  /**
   * Commits the inserts for the non-native tables. Used in the {@link org.apache.hadoop.hive.ql.exec.MoveTask}.
   * @param commitProperties Commit properties which are needed for the handler based commit
   * @param overwrite If this is an INSERT OVERWRITE then it is true
   * @throws HiveException If there is an error during commit
   */
  default void storageHandlerCommit(Properties commitProperties, boolean overwrite) throws HiveException {
    throw new UnsupportedOperationException();
  }
```

Reply:
Sounds good.
Maybe for the first one, `commitWithMoveTask() default true;`?

Reply:
settled on `commitInMoveTask` and `storageHandlerCommit`